src/main/java/build/buf/protovalidate/internal/evaluator/AnyEvaluator.java

```diff
@@ -74,45 +77,42 @@ class AnyEvaluator implements Evaluator {
   }
 
   @Override
-  public ValidationResult evaluate(Value val, boolean failFast) throws ExecutionException {
+  public List<ConstraintViolation.Builder> evaluate(Value val, boolean failFast)
+      throws ExecutionException {
     Message anyValue = val.messageValue();
     if (anyValue == null) {
-      return ValidationResult.EMPTY;
+      return ConstraintViolation.NO_VIOLATIONS;
     }
-    List<Violation> violationList = new ArrayList<>();
+    List<ConstraintViolation.Builder> violationList = new ArrayList<>();
     String typeURL = (String) anyValue.getField(typeURLDescriptor);
     if (!in.isEmpty() && !in.contains(typeURL)) {
-      Violation violation =
-          Violation.newBuilder()
-              .setProto(
-                  build.buf.validate.Violation.newBuilder()
-                      .setRule(IN_RULE_PATH)
-                      .setConstraintId("any.in")
-                      .setMessage("type URL must be in the allow list")
-                      .build())
-              .setFieldValue(val)
-              .setRuleValue(new ObjectValue(IN_DESCRIPTOR, this.inValue))
-              .build();
+      ConstraintViolation.Builder violation =
+          ConstraintViolation.newBuilder()
+              .addAllRulePathElements(getRulePrefixElements())
+              .addAllRulePathElements(IN_RULE_PATH.getElementsList())
+              .addFirstFieldPathElement(getFieldPathElement())
+              .setConstraintId("any.in")
+              .setMessage("type URL must be in the allow list")
+              .setFieldValue(new ConstraintViolation.FieldValue(val))
+              .setRuleValue(new ConstraintViolation.FieldValue(this.inValue, IN_DESCRIPTOR));
```

Comment:
It would be nice to try to wrap some of this up into a helper in `EvaluatorBase`, to de-boiler-plate the various calls that do this.

It looks like they all call `ConstraintViolation.newBuilder().addAllRulePathElements(getRulePrefixElements())`, but the other methods vary slightly. Many also call `.addFirstFieldPathElement(getFieldPathElement())`, but some do not. Anyhow, maybe not worth looking into since they all seem to vary a little -- moving the builder pattern steps into method arguments would certainly make call sites more concise, but potentially at the cost of readability, so maybe not worth doing more here...

Reply:
I am leaning on the other direction (not having `EvaluatorBase`) instead, since I'm on the side of this not being worth it, at least for now. It does feel like there's likely something more elegant to pull out of this, but with all of the edge cases we have right now it's not currently apparent.